codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/model/AWSClientContextParamsTransformer.kt

```diff
@@ -61,6 +61,6 @@ fun ParameterType.toShapeType(): ShapeType? {
     return when (this) {
         ParameterType.STRING -> ShapeType.STRING
         ParameterType.BOOLEAN -> ShapeType.BOOLEAN
-        else -> null
+        ParameterType.STRING_ARRAY -> ShapeType.LIST
```

Comment:
Added `STRING_ARRAY` case and eliminated the `else` so this fails to compile when another `ParameterType` case is added.